src/types.rs

```diff
@@ -50,6 +50,10 @@ pub enum Error {
     /// Failed to (de)serialize a JSON object
     #[error("failed to (de)serialize JSON: {0}")]
     Json(#[from] serde_json::Error),
+    /// Failed to build a Certificate Signing Request
+    #[cfg(feature = "rcgen")]
+    #[error("failed to generate CSR: {0}")]
+    Rcgen(#[from] rcgen::Error),
```

Comment:
Will this make updating rcgen across major versions a semver breaking action for this crate?